Erik extract map component

EHaake opened the pull request:
## Changes
1) Extract MapView into it's own component `Map` so that can be reused and to clean up the PostingDetailScreen.
2) Modify styling so that map is rectangular and wider. I think it works better this way.
3) Hide map if Platform is iOS since it currently crashes. Will need to find a solution for this.

## Tested on
Android Emulator and iOS.

EHaake committed:
Add custom Map Component

EHaake committed:
Try getting custom map component working

Getting 'unknown error' which is very unhelpful

EHaake committed:
Successfully extract point

Refactoring logic currently not working but fine for now

EHaake committed:
Restyle Map to be more rectangular

EHaake committed:
Hide map if Platform is iOS

Expo currently crashes on iOS on any screen with a map, so not rendering
it for now.

seansisson reviewed (dismissed):
Confirmed working on android and it looks great! Approving :)

EHaake committed:
Remove map conditional render

iOS map only seems to crash if the coordinates are invalid. If they are
valid, no crash

EHaake committed:
Add some values to colors and apply to map circle

EHaake committed:
Merge branch 'master' into erik-extract-map-component
